client/signup/config/flows.js

```diff
@@ -220,8 +221,9 @@ const Flows = {
 	 */
 	getABTestFilteredFlow( flowName, flow ) {
 		// Only do this on the main flow
-		// if ( 'main' === flowName ) {
-		// }
+		if ( 'main' === flowName ) {
+			return Flows.getFlows()[ abtest( 'improvedOnboarding' ) ] || flow;
+		}
```

Comment:
@southp 

I think the problem (reverted in #28042) was the the steps were 'reset'. 

The `user` step had already taken place, but we were reinserting it.

<img width="1046" alt="screen shot 2018-10-24 at 5 54 15 pm" src="https://user-images.githubusercontent.com/6458278/47412171-a52cda80-d7b6-11e8-8943-d977b2004ed1.png">

I don't know if this is bullet-proof, but something like this worked for me while debugging:

```
	getABTestFilteredFlow( flowName, flow ) {
		// Only do this on the main flow
		if ( 'main' === flowName ) {
			const abTestFilteredFlow = Flows.getFlows()[ abtest( 'improvedOnboarding' ) ];
			if ( abTestFilteredFlow ) {
				// If we've already completed a step in the flow...
				abTestFilteredFlow.steps = intersection( abTestFilteredFlow.steps, flow.steps );
				return abTestFilteredFlow;
			}
		}
		return flow;
	},

```

Reply:
Right. I didn't notice there is a call to `removeUserStepFromFlow()` here: https://github.com/Automattic/wp-calypso/blob/master/client/signup/config/flows.js#L157 🤦‍♂️

Reply:
Something occurred to me: `intersection` might not be the best choice, since we want to preserve all remaining steps in the ab test flow, and they may or may not match 1:1 with the current flow.

Hope that doesn't confuse things further. I just wanted to make that point before I forgot it myself. :)